### PR TITLE
nixUnstable: 3.0pre20201020_e0ca98c -> 2.4pre20201102_550e11f

### PR DESCRIPTION
**Pull request by grahamc**
###### Motivation for this change

Fixes recursive Nix, nix-copy-closure on drvs, some memory bugs, etc.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/nix/default.nix [hunk 1]
```diff
@@ -32,16 +32,17 @@ common =
       inherit name src;
       version = lib.getVersion name;
 
-      is30 = lib.versionAtLeast version "3.0pre";
-      isExactly30 = lib.versionAtLeast version "2.3" && lib.versionOlder version "3.0";
+
```
Comment (cole-h) [inline pkgs/tools/package-management/nix/default.nix:35]:
```suggestion
```

Comment (grahamc) [reply pkgs/tools/package-management/nix/default.nix:35]: Fixup'd.

**Review by cole-h — APPROVED** [review]
Diff LGTM.

---

Result of `nixpkgs-review pr 102532` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>nix-exec</li>
  </ul>
</details>
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>tests.nixos-functions.nixos-test</li>
    <li>tests.nixos-functions.nixosTest-test</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>fusionInventory</li>
    <li>python37Packages.nixpkgs</li>
  </ul>
</details>
<details>
  <summary>49 packages built:</summary>
  <ul>
    <li>bundix</li>
    <li>cabal2nix</li>
    <li>cachix</li>
    <li>common-updater-scripts</li>
    <li>crystal2nix</li>
    <li>dep2nix</li>
    <li>disnix</li>
    <li>disnixos</li>
    <li>go2nix</li>
    <li>haskellPackages.cachix</li>
    <li>haskellPackages.hercules-ci-agent</li>
    <li>haskellPackages.hocker</li>
    <li>haskellPackages.nix-paths</li>
    <li>hercules-ci-agent</li>
    <li>hydra-unstable</li>
    <li>lispPackages.quicklisp-to-nix</li>
    <li>lispPackages.quicklisp-to-nix-system-info</li>
    <li>nix (nixStable)</li>
    <li>nix-bundle</li>
    <li>nix-direnv</li>
    <li>nix-doc</li>
    <li>nix-du</li>
    <li>nix-index</li>
    <li>nix-pin</li>
    <li>nix-plugins</li>
    <li>nix-prefetch</li>
    <li>nix-prefetch-bzr</li>
    <li>nix-prefetch-cvs</li>
    <li>nix-prefetch-docker</li>
    <li>nix-prefetch-git</li>
    <li>nix-prefetch-hg</li>
    <li>nix-prefetch-scripts</li>
    <li>nix-prefetch-svn</li>
    <li>nix-serve</li>
    <li>nix-update</li>
    <li>nix-update-source</li>
    <li>nixFlakes (nixUnstable)</li>
    <li>nixos-generators</li>
    <li>nixos-shell</li>
    <li>nixpkgs-review</li>
    <li>nixui</li>
    <li>python27Packages.nix-kernel</li>
    <li>python37Packages.nix-kernel</li>
    <li>python37Packages.pythonix</li>
    <li>python38Packages.nix-kernel</li>
    <li>python38Packages.nixpkgs</li>
    <li>python38Packages.pythonix</li>
    <li>vgo2nix</li>
    <li>vulnix</li>
  </ul>
</details>

**Comment by Ma27** [comment]
@grahamc shouldn't it be `2.4pre20201102` as the latest rev in the package is from today?

**Review by Ma27 — APPROVED** [review]
Should also be backported to 20.09 IMHO.

**Comment by cole-h** [comment]
Backported in https://github.com/NixOS/nixpkgs/pull/102538 (ensuring cross-reference).

**Pull request merged**